collect_app/src/test/java/org/odk/collect/android/widgets/DateWidgetTest.java

```diff
@@ -2,65 +2,158 @@
+import static org.odk.collect.android.widgets.support.QuestionWidgetHelpers.promptWithReadOnlyAndQuestionDef;
+import static org.odk.collect.android.widgets.support.QuestionWidgetHelpers.widgetTestActivity;
+
+@RunWith(RobolectricTestRunner.class)
+public class DateWidgetTest {
+    private TestScreenContextActivity widgetActivity;
+    private DateTimeWidgetUtils widgetUtils;
+    private View.OnLongClickListener onLongClickListener;
+
+    private QuestionDef questionDef;
+    private LocalDateTime dateAnswer;
+
+    @Before
+    public void setUp() {
+        widgetActivity = widgetTestActivity();
+
+        questionDef = mock(QuestionDef.class);
+        onLongClickListener = mock(View.OnLongClickListener.class);
+        widgetUtils = mock(DateTimeWidgetUtils.class);
+
+        dateAnswer = DateTimeUtils.getSelectedDate(new LocalDateTime().withDate(2010, 5, 12), LocalDateTime.now());
+    }
+
+    @Test
+    public void usingReadOnlyOption_doesNotShowButton() {
+        DateWidget widget = createWidget(promptWithReadOnlyAndQuestionDef(questionDef));
+        assertEquals(widget.binding.widgetButton.getVisibility(), View.GONE);
+    }
+
+    @Test
+    public void whenPromptIsNotReadOnly_buttonShowsCorrectText() {
+        DateWidget widget = createWidget(promptWithQuestionDefAndAnswer(questionDef, null));
+        assertEquals(widget.binding.widgetButton.getText(), widget.getContext().getString(R.string.select_date));
+    }
+
+    @Test
+    public void getAnswer_whenPromptDoesNotHaveAnswer_returnsNull() {
+        assertThat(createWidget(promptWithQuestionDefAndAnswer(questionDef, null)).getAnswer(), nullValue());
+    }
+
+    @Test
+    public void getAnswer_whenPromptHasAnswer_returnsDate() {
+        DateWidget widget = createWidget(promptWithQuestionDefAndAnswer(questionDef, new DateData(dateAnswer.toDate())));
+        assertEquals(widget.getAnswer().getDisplayText(), new DateData(dateAnswer.toDate()).getDisplayText());
+    }
 
-public class DateWidgetTest extends GeneralDateTimeWidgetTest<DateWidget, DateData> {
+    @Test
+    public void whenPromptDoesNotHaveAnswer_answerTextViewShowsNoDateSelected() {
+        DateWidget widget = createWidget(promptWithQuestionDefAndAnswer(questionDef, null));
+        assertEquals(widget.binding.widgetAnswerText.getText(), widget.getContext().getString(R.string.no_date_selected));
+    }
+
+    @Test
+    public void whenPromptHasAnswer_answerTextViewShowsCorrectDate() {
+        FormEntryPrompt prompt = promptWithQuestionDefAndAnswer(questionDef, new DateData(dateAnswer.toDate()));
+        DatePickerDetails datePickerDetails = DateTimeWidgetUtils.getDatePickerDetails(prompt.getQuestion().getAppearanceAttr());
+        DateWidget widget = createWidget(prompt);
+
+        assertEquals(widget.binding.widgetAnswerText.getText(),
+                DateTimeWidgetUtils.getDateTimeLabel(dateAnswer.toDate(), datePickerDetails, false, widget.getContext()));
+    }
```

Comment:
Hi @grzesiek2010! I had a doubt. Here, I just test whether the date answer shown in `widgetAnswerText` is same as that returned by `DateTimeWidgetUtils.getDateTimeLAbel()` method. I do not write unit tests for all the different cases when `datePickerDetails` are different, and for different formats. Is it fine here just to check fior one case , and have a check for all the other cases in `DateTimeWidgetUtilsTest` class?

Reply:
Yeah I think it's fine